crates/sui-core/src/authority.rs

```diff
@@ -2116,6 +2117,127 @@ impl AuthorityState {
         })
     }
 
+    /// The object ID for gas can be any object ID, even for an uncreated object
+    #[instrument(skip_all)]
+    pub fn call_view_function(
+        &self,
+        view_function: ProgrammableTransaction,
```

Comment:
This is an entire `ProgrammableTransaction`. Why is this function called `call_view_function`? There can be _A LOT_ more going on within this payload.

Reply:
this was an "easy" starting point for discussion (and to build out a simple POC) since there are still a number of semantics around this that are open questions. Some of the open questions listed above may have us align on using ProgrammableTransaction type as-is (maybe with some linting) or require creating something completely different that only allows MoveCalls with Input args.

Reply:
```
struct ViewFunction {
    inputs: Vec<Input>,
    function: MoveCall,
}
```

Would be one such example or we could even not use a MoveCall and create a different struct if we wanted to not need to worry about the semantics around `Input` and `Argument` being conflated here.

Reply:
I would personally favour keeping this as a programmable transaction, so that we don't end up inventing a new structure that resembles commands, inputs, arguments, and results, but we call this something different to acknowledge that it could be doing something more than just calling a list of view functions.

Reply:
+1 to keeping this a programmable transaction.

At the risk of being too terse and/or general I kinda like the idea of just calling this a `view` (or maybe a `view_transaction`?).